asserts/database.go

```diff
@@ -27,22 +27,34 @@ import (
 	"fmt"
 	"os"
 	"path/filepath"
+	"strings"
+	"time"
 
 	"golang.org/x/crypto/openpgp/packet"
 
 	"github.com/ubuntu-core/snappy/helpers"
 )
 
+// PublicKey is a public key as used by the assertion database.
+type PublicKey interface {
+	PublicKeyPrim
+	// IsValidAt returns whether the public key is valid at 'when' time
+	IsValidAt(when time.Time) bool
+}
+
 // DatabaseConfig for an assertion database.
 type DatabaseConfig struct {
 	// database backstore path
 	Path string
+	// trusted keys maps authority-ids to list of trusted keys.
+	TrustedKeys map[string][]PublicKey
 }
 
 // Database holds assertions and can be used to sign or check
 // further assertions.
 type Database struct {
 	root string
+	cfg  *DatabaseConfig
```

Comment:
I suggest making this a value rather than a reference, and copying the incoming configuration when opening (`db.cfg = *cfg`).